packages/liveblocks-core/src/room.ts

```diff
@@ -1140,27 +1142,44 @@ export function createRoom<
     comments: makeEventSource<CommentsEventServerMsg>(),
   };
 
-  function sendMessages(
-    messageOrMessages:
-      | ClientMsg<TPresence, TRoomEvent>
-      | ClientMsg<TPresence, TRoomEvent>[]
+  async function httpSend(
+    authTokenOrPublicApiKey: string,
+    roomId: string,
+    nonce: string,
+    messages: ClientMsg<TPresence, TRoomEvent>[]
   ) {
-    const message = JSON.stringify(messageOrMessages);
-    if (config.unstable_fallbackToHTTP) {
+    const baseUrl = new URL(config.liveblocksServer);
+    baseUrl.protocol = "https";
+    const url = new URL(
+      `/v2/c/rooms/${encodeURIComponent(roomId)}/send-message`,
+      baseUrl
+    );
+    const fetcher = config.polyfills?.fetch || /* istanbul ignore next */ fetch;
+    return fetcher(url.toString(), {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${authTokenOrPublicApiKey}`,
+      },
+      body: JSON.stringify({ nonce, messages }),
+    });
+  }
```

Comment:
Key change here is twofold:

- Using `/v2/c/` now, instead of `/v2/` (L1154)
- Sending the `nonce` in the payload, alongside `messages` (previously `messages` _was_ the payload) (L1164)